enterprise/coderd/prebuilds/metricscollector_test.go

```diff
@@ -0,0 +1,331 @@
+package prebuilds_test
+
+import (
+	"fmt"
+	"slices"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"tailscale.com/types/ptr"
+
+	"github.com/prometheus/client_golang/prometheus"
+	prometheus_client "github.com/prometheus/client_model/go"
+
+	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/quartz"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbgen"
+	"github.com/coder/coder/v2/coderd/database/dbtestutil"
+	agplprebuilds "github.com/coder/coder/v2/coderd/prebuilds"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/enterprise/coderd/prebuilds"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestMetricsCollector(t *testing.T) {
+	t.Parallel()
+
+	if !dbtestutil.WillUsePostgres() {
+		t.Skip("this test requires postgres")
+	}
+
+	type metricCheck struct {
+		name      string
+		value     *float64
+		isCounter bool
+	}
+
+	type testCase struct {
+		name            string
+		transitions     []database.WorkspaceTransition
+		jobStatuses     []database.ProvisionerJobStatus
+		initiatorIDs    []uuid.UUID
+		ownerIDs        []uuid.UUID
+		metrics         []metricCheck
+		templateDeleted []bool
+		eligible        []bool
+	}
+
+	tests := []testCase{
+		{
+			name:         "prebuild provisioned but not completed",
+			transitions:  allTransitions,
+			jobStatuses:  allJobStatusesExcept(database.ProvisionerJobStatusPending, database.ProvisionerJobStatusRunning, database.ProvisionerJobStatusCanceling),
+			initiatorIDs: []uuid.UUID{agplprebuilds.SystemUserID},
+			ownerIDs:     []uuid.UUID{agplprebuilds.SystemUserID},
+			metrics: []metricCheck{
+				{"coderd_prebuilt_workspaces_created_total", ptr.To(1.0), true},
+				{"coderd_prebuilt_workspaces_claimed_total", ptr.To(0.0), true},
+				{"coderd_prebuilt_workspaces_failed_total", ptr.To(0.0), true},
+				{"coderd_prebuilt_workspaces_desired", ptr.To(1.0), false},
+				{"coderd_prebuilt_workspaces_running", ptr.To(0.0), false},
+				{"coderd_prebuilt_workspaces_eligible", ptr.To(0.0), false},
+			},
+			templateDeleted: []bool{false},
+			eligible:        []bool{false},
```

Comment:
I'm not seeing any test-cases where we'd need slices for deleted/eligible and it probably isn't too compatible with the way expected metrics are defined. Consider simplifying (helps reduce complexity of test body too)?